openhands/controller/agent_controller.py

```diff
@@ -312,6 +312,15 @@ async def _handle_message_action(self, action: MessageAction) -> None:
                 str(action),
                 extra={'msg_type': 'ACTION', 'event_source': EventSource.USER},
             )
+            # Extend max iterations when the user sends a message
+            if self._initial_max_iterations is not None:
+                self.state.max_iterations = (
+                    self.state.iteration + self._initial_max_iterations
+                )
+                self.log(
+                    'info',
+                    f'Extended max iterations to {self.state.max_iterations} after user message',
```

Comment:
I think this will also happen with evals, if there's a fake user message? Pretty sure it's indeed a MessageAction with source USER.

Reply:
@xingyaoww how concerned are you about this?

Reply:
🤔 hmm i think this will definitely breka eval in certain way..

I wonder if we can just add an additonal config setting `extend_max_iteration_after_user_message: bool` in `AppConfig` and default it to False? And in `listen_socket.py` / session which is used for application, we override them to True? This way it won't break the eval behavior while also giving us the max iteration extension behavior.

Reply:
@xingyaoww Actually, we already have [headless_mode](https://github.com/All-Hands-AI/OpenHands/blob/7ef6fa666d61390194340133e154a33b4e1f6b26/openhands/controller/agent_controller.py#L90), which only the agent session sets to False. We can only extend max iterations when headless_mode is False.